### DISCUSSION TITLE
Multiple devices

### DISCUSSION THREAD

**Discussion by n9yty** [BorisBrock/Sunalyzer#82] [Ideas]
Hi @BorisBrock,

I was looking at the pull request to support multiple devices.

What can be done to revise it so that it meets your expectations and can be merged in?  I would like to add support for the Emporia monitoring system, but it would depend on this pull request being incorporated.

I don't know if the original submitter is around to do revisions, so I don't know what we can do to "fix" the request so it will merge now and then revise things. I'm not super savvy on git beyond simple use cases so I don't know how this would be handled.

**Comment by BorisBrock** [comment]
Hi @n9yty 
for proper multi device support, I see the following requirements (just a quick brainstorming):

- Two distinct device types: 
  - Smart Meters (to measure feed in/consumption from grid)
  - "Production Devices" (e.g. solar inverters, a simple ShellyPlug etc.)
- The configuration can contain at most one smart meter
- The configuration can contain an arbitrary amount of production devices, each having user defined name for the GUI
- The data shall be captured, stored and displayed in the front end...
  - ...for each device individually
  - ...summed up as single, combined value
- There shall be no dependencies between these device types, e.g. different brands of production devices and smart meters shall be combineable
- The whole system should work without a smart meter, or without production devices. In that case the GUI should adapt accordingly

So this would require a bit of work... ;-)
